### ISSUE TITLE
Make nools tests run when changes are made

### ISSUE THREAD

**Issue by SCdF** [medic/medic-webapp#2504]
In #1836 we created a framework to write nools tests, and in #2503 we wrote tests for LG's rules using that framework.

In this ticket we should build / setup something that ensures that nools tests are run when the rules or their tests change. This would ideally happen automatically for all tests in medic-projects, though could require that there is some initial work required for each project before they are automatically tested.

One way of doing this would be submitting medic-projects to travis, and having some kind of script that searches for test directories (or has a list of them which you add to) and runs them all. This would require that medic-projects goes to Travis though, I'm unsure if that's OK. The other alternative is that we use another CI tool, one that we run privately (I believe we have a Jenkins?)

This ticket is complete when changes that are made to tests or the rules that are being tested results in those tests being automatically run in a CI environment.

**Comment by alxndrsn** [comment]
For testing private repos on Travis, we'll need Travis Pro (as per https://docs.travis-ci.com/user/travis-pro/).

**Comment by nice-snek** [comment]
hi frand @sglangevin, please [triage](https://github.com/medic/medic-docs/blob/master/md/dev/workflow.md#triaging-old-issues) before the end of this sprint. :parrot_conga::parrot_conga::parrot_conga:

**Comment by sglangevin** [comment]
This seems to be blocked on #2503. Marking as blocked for now. Seems like #2503 is a good next step in building out a test framework for Nools rules (tasks and targets).

**Comment by SCdF** [comment]
@sglangevin FWIW I'd say the testing framework in #1836 was really more of a prototype: I don't think a new project could use it today for example. So this ticket is also blocked on us truly spending the time to build something people could use, and then either filling out LG's tests as examples, or not getting the time to do it and just letting other projects go it alone.

**Comment by sglangevin** [comment]
I'd like to see us build out a testing framework for configurations, so I'm going to leave this one as blocked for now and add the Testing label so that it's easy to find later. Sounds like we'd need to revisit #1836 first and perhaps open a new issue for building out something that people can use. #2503 sounded like it was referring to building out the tests for LG's rules, which I assumed would also include building more of the framework.

**Comment by alxndrsn** [comment]
I think this is done.

**Comment by ngaruko** [comment]
Is this the same as #1836 @alxndrsn @SCdF ?  Are these tests run on Travis or elsewhere?

**Comment by SCdF** [comment]
@ngaruko not completely. This was about having them run on changes. i..e, having travis run automatically on change

**Comment by newtewt** [comment]
@SCdF  is there more to this than having a build triggered by travis when a PR is created?

**Comment by SCdF** [comment]
@newtewt not just when a PR is created, but also on Master in general.

Having said that, I don't really think this needs any kind of AT, I think it can just be closed.
